src/ocean/text/arguments/DateIntervalArgs.d

```diff
@@ -0,0 +1,429 @@
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range now 2014-03-09");
+    date_args.processArgs(args);
+    timeToUnixTime("2014-03-09 23:59:59", dummy_time, dummy_conv);
+
+    Test.test!("==")(date_args.begin(), time(null));
+    Test.test!("==")(date_args.end(), dummy_time);
+
+    /// When both dates are provided
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range 2014-03-09 2014-03-09");
+    date_args.processArgs(args);
+
+    timeToUnixTime("2014-03-09 00:00:00", dummy_time, dummy_conv);
+    Test.test!("==")(date_args.begin(), dummy_time);
+
+    timeToUnixTime("2014-03-09 23:59:59", dummy_time, dummy_conv);
+    Test.test!("==")(date_args.end(), dummy_time);
+
+    /// Using timestamps
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range 1000 1200");
+    date_args.processArgs(args);
+
+    Test.test!("==")(date_args.begin(), 1000);
+    Test.test!("==")(date_args.end(), 1200);
+
+    /// Using interval starting with time duration
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range 1m 1000");
+    date_args.processArgs(args);
+
+    Test.test!("==")(date_args.begin(), 940);
+    Test.test!("==")(date_args.end(), 1000);
+
+    /// Using interval ending with time duration
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range 1000 1m");
+    date_args.processArgs(args);
+
+    Test.test!("==")(date_args.begin(), 1000);
+    Test.test!("==")(date_args.end(), 1060);
+
+    /// Using two  time durations
+    args = new Arguments;
+    date_args = new DateIntervalArgs(interval);
+
+    date_args.setupArgs!(false)(args);
+    args.parse("--time-range 1m 1m");
+    date_args.processArgs(args);
+
+    Test.test!("==")(date_args.begin(), time(null) - 60);
+    Test.test!("==")(date_args.end(), time(null) + 60);
+
+}
+
+/******************************************************************************
+
+    Converts a string to seconds
+
+    Params:
+        value = a string containing an integer and the first leter of a time
+            unit. The supported units are minutes, hours and days.
+
+    Returns:
+        The interval in seconds
+
+******************************************************************************/
+
+public long parseTimeInterval ( cstring value )
+{
+    if ( value == "" )
+    {
+        return 0;
+    }
+
+    long result;
+    long tempValue;
+
+    char unit = value[value.length - 1];
+
+    if ( !toLong(value[0..$-1], tempValue) )
+    {
+        throw new Exception("The provided time interval has an invalid value.");
+    }
+
+    switch ( unit )
```

Comment:
this looks like it'd be easy to support seconds. Any reason not too?

Reply:
no reason. I just added it.